backend/typescript/rest/authRoutes.ts

```diff
@@ -103,6 +103,33 @@ authRouter.post("/register", registerRequestValidator, async (req, res) => {
   }
 });
 
+/**
+ * Sends the user an email to reset their password. Used when the user has forgotten their password.
+ * @returns bool - true if successfully sent the email, false otherwise
+ */
+authRouter.post("/forgotPassword", async (req, res) => {
+  try {
+    // Find the user record with that email. If none found, error will be caught
+    const firebaseUserUid = await authService.getFirebaseUserIdByEmail(
+      req.body.email,
+    );
+
+    // Set the user's password to the random access code. This is needed on the front end to verify the user
+    const accessCode = await authService.setTemporaryUserPassword(
```

Comment:
Do we want to change a user's password if they forget it? ie consider the case where they click forgot password, but immediately remember their old password

Reply:
resolved for now as discussed